### PR TITLE
M0078DDG-457 Document tracing method invocation.

### PR DESCRIPTION
**Pull request by dalgassimou**

### CODE REVIEW

File: PlatformDeveloperGuide/advanceTrace.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+But calling these hook functions each time a method is invoked can slow the whole application,
+so these hook functions is not activated by default even when the trace is enabled and started.
+
+To activate them you need:
+
+- Enable and start the trace see here https://docs.microej.com/en/latest/ApplicationDeveloperGuide/trace.html?highlight=trace#event-recording
+- Tell the Third-party linker program to redirect all calls to ``LLMJVM_invoke_method`` and ``LLMJVM_return_method`` symbols to respectively ``LLMJVM_invoke_method_with_trace`` and ``LLMJVM_return_method_with_trace`` symbols.
+
+
+Platform which use LD as linker
```
Comment (frivieremicroej) [inline PlatformDeveloperGuide/advanceTrace.rst:25]:
```suggestion
Platform using GNU LD linker
```

File: PlatformDeveloperGuide/advanceTrace.rst [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+.. code-block:: sh
+
+    --require-defined=LLMJVM_invoke_method_with_trace
+    --defsym=LLMJVM_invoke_method=LLMJVM_invoke_method_with_trace
+    --require-defined=LLMJVM_return_method_with_trace
+    --defsym=LLMJVM_return_method=LLMJVM_return_method_with_trace
+
+
+
+Platform which use ILINK as linker (IAR)
```
Comment (frivieremicroej) [inline PlatformDeveloperGuide/advanceTrace.rst:39]:
```suggestion
Platform using IAR ILINK linker
```

File: PlatformDeveloperGuide/advanceTrace.rst [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+========================================
+
+Pass the following options to the ILINK linker program
```
Comment (frivieremicroej) [inline PlatformDeveloperGuide/advanceTrace.rst:42]:
```suggestion
Add the following options to the ILINK linker command line:
```

File: PlatformDeveloperGuide/advanceTrace.rst [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+To activate them you need:
+
+- Enable and start the trace see here https://docs.microej.com/en/latest/ApplicationDeveloperGuide/trace.html?highlight=trace#event-recording
+- Tell the Third-party linker program to redirect all calls to ``LLMJVM_invoke_method`` and ``LLMJVM_return_method`` symbols to respectively ``LLMJVM_invoke_method_with_trace`` and ``LLMJVM_return_method_with_trace`` symbols.
+
+
+Platform which use LD as linker
+===============================
+
+Pass the following options to the LD linker program
```
Comment (frivieremicroej) [inline PlatformDeveloperGuide/advanceTrace.rst:28]:
```suggestion
Add the following options to the LD linker command line
```

**Review by frivieremicroej — CHANGES_REQUESTED** [review]

File: PlatformDeveloperGuide/advanceTrace.rst [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+Advanced Event Tracing
+######################
+
+Principle
+=========
+
+MicroEJ allows method execution to be profiled, two new hooks functions are used for that:
+
+- ``LLMJVM_MONITOR_IMPL_on_invoke_method`` called at the start of the method invocation.
+- ``LLMJVM_MONITOR_IMPL_on_return_method`` called at the end of the method invocation.
```
Comment (jerome-leroux) [inline PlatformDeveloperGuide/advanceTrace.rst:16]:
```suggestion
- ``LLMJVM_MONITOR_IMPL_on_return_method`` called when returning from the invoked method.
```

File: PlatformDeveloperGuide/advanceTrace.rst [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+But calling these functions each time a method is invoked can slow the whole application,
+so these functions are not called by default when event tracing is enabled and started.
```
Comment (jerome-leroux) [inline PlatformDeveloperGuide/advanceTrace.rst:19]:
```suggestion
Calling these functions each time a method is invoked can slow down the whole application,
so these functions are not called by default when event tracing is enabled and started.
```

File: PlatformDeveloperGuide/advanceTrace.rst [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+.. note::
+   This feature is available in the Architecture versions 7.17.0 or higher for the Applications deployed on hardware devices (not on Simulator).
+
+To activate them you need:
```
Comment (jerome-leroux) [inline PlatformDeveloperGuide/advanceTrace.rst:24]:
```suggestion
To activate them, you need to follow these steps:
```

File: PlatformDeveloperGuide/advanceTrace.rst [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+..  _Advanced-Event-Tracing:
+
+Advanced Event Tracing
+######################
+
+Principle
+=========
+
+MicroEJ allows method execution to be profiled, two new hooks functions are used for that:
```
Comment (jerome-leroux) [inline PlatformDeveloperGuide/advanceTrace.rst:13]:
```suggestion
MicroEJ allows method execution to be profiled. The following two new hooks functions are used for that:
```

File: PlatformDeveloperGuide/advanceTrace.rst [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+.. note::
+   This feature is available in the Architecture versions 7.17.0 or higher for the Applications deployed on hardware devices (not on Simulator).
+
+To activate them you need:
+
+- :ref:`Enable and start event tracing <event_enable_recording>`
+- Tell the Third-party linker program to redirect all calls to ``LLMJVM_invoke_method`` and ``LLMJVM_return_method`` symbols to respectively ``LLMJVM_invoke_method_with_trace`` and ``LLMJVM_return_method_with_trace`` symbols.
+
+Platform using GNU LD linker
```
Comment (jerome-leroux) [inline PlatformDeveloperGuide/advanceTrace.rst:29]:
```suggestion
Platforms using GNU LD linker
```

File: PlatformDeveloperGuide/advanceTrace.rst [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+.. code-block:: sh
+
+    --require-defined=LLMJVM_invoke_method_with_trace
+    --defsym=LLMJVM_invoke_method=LLMJVM_invoke_method_with_trace
+    --require-defined=LLMJVM_return_method_with_trace
+    --defsym=LLMJVM_return_method=LLMJVM_return_method_with_trace
+
+
+Platform using IAR ILINK linker
```
Comment (jerome-leroux) [inline PlatformDeveloperGuide/advanceTrace.rst:42]:
```suggestion
Platforms using IAR ILINK linker
```

File: PlatformDeveloperGuide/advanceTrace.rst [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+To activate them you need:
+
+- :ref:`Enable and start event tracing <event_enable_recording>`
+- Tell the Third-party linker program to redirect all calls to ``LLMJVM_invoke_method`` and ``LLMJVM_return_method`` symbols to respectively ``LLMJVM_invoke_method_with_trace`` and ``LLMJVM_return_method_with_trace`` symbols.
+
+Platform using GNU LD linker
+============================
+
+Add the following options to the LD linker command line
```
Comment (jerome-leroux) [inline PlatformDeveloperGuide/advanceTrace.rst:32]:
```suggestion
Add the following options to the LD linker command line:
```

**Review by jerome-leroux — CHANGES_REQUESTED** [review]

**Review by jerome-leroux — APPROVED** [review]

File: ApplicationDeveloperGuide/trace.rst [hunk 12]
```diff
@@ -136,16 +136,29 @@ It only prints the ID of the recorded event followed by the values given in para
...
-MicroEJ provides an implementation that redirects the events to :ref:`systemview` tool, 
+MicroEJ Corp. provides an implementation that redirects the events to :ref:`systemview` tool, 
 the real-time recording and visualization tool from `Segger <https://www.segger.com/>`_. It is perfect for a finer understanding of the runtime behavior by showing events sequence and duration.
 
 A implementation example for the `NXP OM13098 development board <https://www.nxp.com/products/processors-and-microcontrollers/arm-microcontrollers/general-purpose-mcus/lpc54000-cortex-m4-/lpcxpresso54628-development-board:OM13098>`_ with SystemView support is available `here <https://developer.microej.com/packages/referenceimplementations/U3OER/2.0.1/OM13098-U3OER-fullPackaging-eval-2.0.1.zip>`__.
+
 Please contact :ref:`our support team <get_support>` for more information about how to integrate this Platform module.
-    
+
+Advanced Event tracing
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/trace.rst:146]:
```suggestion
Advanced Event Tracing
```

File: PlatformDeveloperGuide/advanceTrace.rst [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+
+..  _Advanced-Event-Tracing:
+
+Advanced Event Tracing
+######################
+
+Principle
+=========
+
+MicroEJ allows method execution to be profiled. The following two new hooks functions are used for that:
```
Comment (frivieremicroej) [inline PlatformDeveloperGuide/advanceTrace.rst:13]:
```suggestion
MicroEJ Core Engine allows method execution to be profiled. The following two new hooks functions are used for that:
```

File: PlatformDeveloperGuide/advanceTrace.rst [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+
+- ``LLMJVM_MONITOR_IMPL_on_invoke_method`` called at the start of the method invocation.
+- ``LLMJVM_MONITOR_IMPL_on_return_method`` called when returning from the invoked method.
+
+Calling these functions each time a method is invoked can slow down the whole application,
```
Comment (frivieremicroej) [inline PlatformDeveloperGuide/advanceTrace.rst:18]:
```suggestion
Calling these functions each time a method is invoked will slow down the application execution,
```

File: PlatformDeveloperGuide/advanceTrace.rst [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+so these functions are not called by default when event tracing is enabled and started.
+
+.. note::
+   This feature is available in the Architecture versions 7.17.0 or higher for the Applications deployed on hardware devices (not on Simulator).
+
+To activate them, you need to follow these steps:
+
+- Enable and start the trace see here https://docs.microej.com/en/latest/ApplicationDeveloperGuide/trace.html?highlight=trace#event-recording
```
Comment (frivieremicroej) [inline PlatformDeveloperGuide/advanceTrace.rst:26]: turn to internal link

**Review by frivieremicroej — CHANGES_REQUESTED** [review]

**Review by frivieremicroej — APPROVED** [review]

**Pull request merged**